api/query/cache/monitor/monitor.go

```diff
@@ -114,19 +116,36 @@ func (m *indexMonitor) Start() error {
 		return err
 	}
 
+	m.idx.SetIngestChan(m.runIngested)
+
+	reqs := uint(0)
 	for {
 		if !m.isRunning {
 			return errStopped
 		}
 
-		heapBytes := m.rt.GetHeapBytes()
-		if heapBytes > m.maxHeapBytes {
-			m.logger.Errorf("Out of memory: %d > %d", heapBytes, m.maxHeapBytes)
-			m.idx.EvictRuns(m.percent)
-		} else {
-			m.logger.Debugf("Monitor: %d heap-allocated bytes OK", heapBytes)
+		timer := make(chan bool, 1)
```

Comment:
No need to specify the capacity 1.

Reply:
The capacity of 1 ensures that a goroutine that writes to `timer` can exit (and be cleaned up) immediately, rather than waiting for the `select` to choose `timer`.